src/core/Target_file.ml

```diff
@@ -0,0 +1,19 @@
+type provided_target_language_info =
+  (* The default *)
+  | NoSpecialTargeting
+  (* When `--scan-unknown-extensions` is provided, explicitly passed files
+     are scanned with all languages in the rule *)
+  | ExplicitFileScanUnknownExtension (* | TODO Lang for -lang? *)
+[@@deriving show]
```

Comment:
Can we use underscores for spaces and use capitals only for acronyms e.g. `No_special_targeting` or `HTTP_client`? I never liked camelCase because I find it pointlessly hard to read.

Reply:
I'm not sure if it's relevant historically but `int_of_string` can be typed on a French keyboard without hitting Shift. It makes it easier to type than on an English keyboard, and easier than intOfString on any keyboard. Anyway, I clearly have difficulties reading words that are not separated by some other character (space, underscore, dash, period, ...).

Reply:
Happy to go with whatever! Doesn't make a difference to me